bundles/org.openhab.binding.iaqualink/src/main/java/org/openhab/binding/iaqualink/internal/handler/IAqualinkHandler.java

```diff
@@ -0,0 +1,450 @@
+        pollFuture = scheduler.scheduleWithFixedDelay(() -> {
+            try {
+                pollController();
+            } catch (Exception e) {
+                logger.debug("Exception during poll", e);
+            }
+        }, initalDelay, refresh, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Stops/clears this thing's polling future
+     */
+    private void clearPolling() {
+        ScheduledFuture<?> localFuture = pollFuture;
+        if (localFuture != null && !localFuture.isCancelled()) {
+            logger.trace("Canceling future");
+            localFuture.cancel(true);
+        }
+    }
+
+    /**
+     * Poll the controller for updates.
+     */
+    private void pollController() {
+        try {
+
+            JsonElement home = client.getHomeJson(serialNumber, sessionId);
+            JsonArray homeScreen = home.getAsJsonObject().getAsJsonArray("home_screen");
+            if (homeScreen != null) {
+                homeScreen.forEach(element -> {
+                    element.getAsJsonObject().entrySet().forEach(entry -> {
+                        updatedState(entry.getKey(), entry.getValue().getAsString());
+                        if (entry.getKey().endsWith("_heater")) {
+                            updatedState(entry.getKey() + "_status", entry.getValue().getAsString());
+                        }
+                    });
+                });
+            }
+
+            OneTouch[] oneTouches = client.getOneTouch(serialNumber, sessionId);
+            for (OneTouch ot : oneTouches) {
+                updatedState(ot.getName(), ot.getState());
+            }
+
+            Auxiliary[] auxes = client.getAux(serialNumber, sessionId);
+            for (Auxiliary aux : auxes) {
+                updatedState(aux.getName(), aux.getState());
+            }
+
+            if (firstRun) {
+                updateChannels(auxes, oneTouches);
+            }
+
+            firstRun = false;
+
+            if (getThing().getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+
+        } catch (InterruptedException e) {
+            logger.debug("polling interupted", e);
+        } catch (IOException e) {
+            logger.debug("Exception polling", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (NotAuthorizedException e) {
+            logger.debug("Authorization Exception during polling", e);
+            clearPolling();
+            configure();
+        }
+    }
+
+    /**
+     * Update an channels state only if the value of the channel has changed since our last poll.
+     *
+     * @param name
+     * @param value
+     */
+    private void updatedState(String name, String value) {
+        logger.trace("updatedState {} : {}", name, value);
+        Channel channel = getThing().getChannel(name);
+        if (channel != null) {
+            State state = toState(channel.getAcceptedItemType(), value);
```

Comment:
You do mention that `toState` throws a `NumberFormatException` but you don't handle it here and you don't specify to throw it further

Reply:
good catch!